pkg/controller/registry/resolver/resolver.go

```diff
@@ -26,9 +26,10 @@ type OperatorResolver interface {
 }
 
 type SatResolver struct {
-	cache cache.OperatorCacheProvider
-	log   logrus.FieldLogger
-	pc    *predicateConverter
+	cache                     cache.OperatorCacheProvider
+	log                       logrus.FieldLogger
+	pc                        *predicateConverter
+	systemConstraintsProvider solver.ConstraintProvider
```

Comment:
Now that Eric introduced the `predicateConverter`, would like to see if this `systemConstraintsProvider` can be nested under that but it doesn't seem to fit. Another field in SatResolver is a bit messy but it is not too bad if we want to get rid of it later on. Just a thought here. No blocker.